ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/ConnectorDescriptionImpl.java

```diff
@@ -40,18 +42,20 @@ public ConnectorDescriptionImpl(
   ) {
     this.name = Objects.requireNonNull(name, "name");
     this.className = Objects.requireNonNull(className, "className");
-    this.sources = Objects.requireNonNull(sources, "sources");
-    this.topics = Objects.requireNonNull(topics, "topics");
+    this.sources = ImmutableList.copyOf(Objects.requireNonNull(sources, "sources"));
+    this.topics = ImmutableList.copyOf(Objects.requireNonNull(topics, "topics"));
     this.type = Objects.requireNonNull(type, "type");
     this.state = Objects.requireNonNull(state, "state");
   }
 
   @Override
+  @SuppressFBWarnings(value = "EI_EXPOSE_REP", justification = "sources is ImmutableList")
   public List<String> sources() {
```

Comment:
Wondering if we should change the return type to `ImmutableList<String>` ?

Some classes specify the interfaces (`List`, `Map`, `Set`) while others use `ImmutableXxx`. Seems better to use `ImmutableXxx` IMHO but wanted to clarify first. If we don't use `ImmutableXxx` callers might need to make more (unnecessary) deep-copies?

If we want to stick with the interface, we should follow this pattern consistently though and should update the classes that specify `ImmutableXxx` as return type.

Reply:
Not sure if I'm understanding this correctly, but right now none of the classes in the java-api-client module are using ImmutableXXX outside of tests. The interface is pretty consistent now, so if we just change all the constructors to make copies of `List/Set/Map` and getters to return copies, would that resolve the warning?

Reply:
It a general question across the whole code base. I have seen both cases. It's nota about this module in particular.

Most of the time, we use the native Java interfaces `List` etc, but sometime we use `ImmutableList`.

Reply:
Cf https://github.com/confluentinc/ksql/pull/7817/files#r671942538

Reply:
I would vote on using ImmutableList across the rest of the code base, but this specific module is user facing, so I think we should keep this API as is.